[exir] Enable dict for sym shape eval pass

larryliu0820 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #4872

Allow user to pass in a mapping from method name to sym shape eval pass.
This is required by llava vision encoder.


Test:

```
python examples/models/llava/export_llava.py

...

[INFO 2024-08-23 11:21:26,379 export_llava.py:224] Required memory for activation in bytes: [0, 193090816]
[INFO 2024-08-23 11:21:26,379 export_llava.py:224] Required memory for activation in bytes: [0, 2539285504]
[INFO 2024-08-23 11:21:26,379 export_llava.py:224] Required memory for activation in bytes: [0, 573440]
[INFO 2024-08-23 11:21:27,614 export_llava.py:280] Exported ExecuTorch program to llava_combined_xnnpack.pte
```

Differential Revision: [D61728068](https://our.internmc.facebook.com/intern/diff/D61728068)

larryliu0820 commented:
@larryliu0820 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D61728068).

lucylq reviewed (approved):
Lgtm!
Let's update the type hint for sym_shape_eval_pass in ExecutorchBackendConfig as well

larryliu0820 committed:
Enable dict for sym shape eval pass (#4872)

Summary:


Allow user to pass in a mapping from method name to sym shape eval pass.
This is required by llava vision encoder.


Test:

```
python examples/models/llava/export_llava.py

...

[INFO 2024-08-23 11:21:26,379 export_llava.py:224] Required memory for activation in bytes: [0, 193090816]
[INFO 2024-08-23 11:21:26,379 export_llava.py:224] Required memory for activation in bytes: [0, 2539285504]
[INFO 2024-08-23 11:21:26,379 export_llava.py:224] Required memory for activation in bytes: [0, 573440]
[INFO 2024-08-23 11:21:27,614 export_llava.py:280] Exported ExecuTorch program to llava_combined_xnnpack.pte
```

imported-using-ghimport

Test Plan: Imported from OSS

Reviewed By: lucylq

Differential Revision: D61728068

Pulled By: larryliu0820
